switchsdk/src/main/java/net/gini/switchsdk/camera/Camera1.java

```diff
@@ -129,6 +129,11 @@ public void surfaceChanged(final SurfaceHolder surfaceHolder, final int i, final
                     parameters.setFocusMode(FOCUS_MODE_CONTINUOUS_PICTURE);
                 }
 
+                final List<String> supportedFlashModes = parameters.getSupportedFlashModes();
```

Comment:
Isn't it a little strange to deal with flash modes in a `surfaceChanged` method. Which, btw, is really long?